### PR TITLE
[CONSVC-2122] fix: handle empty subdivisions in geolocation parsing

### PR DESCRIPTION
**Pull request by ncloudioj**
This fixes [CONSVC-2122](https://mozilla-hub.atlassian.net/browse/CONSVC-2122)

### CODE REVIEW

File: merino/middleware/geolocation.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         scope[ScopeKey.GEOLOCATION] = (
             Location(
                 country=record.country.iso_code,
-                region=record.subdivisions[0].iso_code,
+                region=record.subdivisions[0].iso_code if record.subdivisions else None,
```
Comment (ncloudioj) [inline merino/middleware/geolocation.py:66]: FWIW, `record.subdivisions` is of `tuple` type.

Comment (Trinaa) [inline merino/middleware/geolocation.py:66]: Should we expand this mitigation to country, city and dma record values?

I think the most sustainable way to test this, would be to find IP addresses in maxmind DB that doesn't resolve a subdivision. Patching has risk with maxmind DB updates.

**Review by Trinaa — APPROVED** [review]
Do you create tasks for fixes like this in JIRA?

**Comment by ncloudioj** [comment]
> Do you create tasks for fixes like this in JIRA?

Yes, I did. See the PR comment.

File: merino/middleware/geolocation.py (hunk 1 above)
Comment (ncloudioj) [reply merino/middleware/geolocation.py:66]: > Should we expand this mitigation to country, city and dma record values?

I've double-checked the other fields, they are all present in `record` and having `None` as fallback value if the info is unavailable. Only the `record.subdivisions` is of `tuple` type so the check is necessary.

Re: the test case for this, agreed, I real IP that doesn't resolve subdivisions would be ideal. I will poke around to see if I can find some for this.

**Pull request merged**